### PR TITLE
Multiple changes

### PR DESCRIPTION
**Pull request by kevinraddatz**
* add the missing GetCompositeSchedule feature
* add some missing create(default)Request/Confirmation methods
* mark the ZeroArgsConstructor as deprecated in favor of the new Constructor with all required arguments
* Replace Calendar with ZonedDateTime (BREAKING CHANGE in API)
* add a missing dependency for SOAP (didn't want to compile without :thinking:

### CODE REVIEW

**Comment by TVolden** [comment]
Hi @kevinraddatz,

Thanks for your PR, it's a big one! 😃 

It's going to take me some time to review, but I'll get to it as soon as I have time.

**Comment by TVolden** [comment]
Sorry for the slow pace. It's a busy week for me, but I'll probably have time monday.

**Comment by kevinraddatz** [comment]
No problem, take your time :)

File: ocpp-v1_6/src/main/java/eu/chargetime/ocpp/feature/profile/ServerSmartChargingProfile.java [hunk 1]
```diff
@@ -45,9 +47,9 @@
 
     public ServerSmartChargingProfile() {
         features = new HashSet<>();
-        features.add(new SetChargingProfileFeature(this));
-        features.add(new GetCompositeScheduleFeature(this));
-        features.add(new ClearChargingProfileFeature(this));
+        features.add(new ClearChargingProfileFeature(null));
+        features.add(new GetCompositeScheduleFeature(null));
+        features.add(new SetChargingProfileFeature(null));
```
Comment (TVolden) [inline ocpp-v1_6/src/main/java/eu/chargetime/ocpp/feature/profile/ServerSmartChargingProfile.java]: Why null?

**Comment by kevinraddatz** [comment]
I set them to null as it seemed like "this" is only required on features that need to be handled by an event handler. The ReserveNow feature for example does not need to be handled on the server so I thought it can be null. 
The ServerCoreFeature already had some features set to null, so I thought to continue with it

**Comment by TVolden** [comment]
I get an error when I try to build and test it with maven. Have you tried that?
I'm going to try, to get the CI to work again, so we can get it's report.

**Comment by TVolden** [comment]
Check it out: https://travis-ci.org/ChargeTimeEU/Java-OCA-OCPP/jobs/619203910
```log=1697
unpackPayload_aCalendarPayload_returnsTestModelWithACalendar(eu.chargetime.ocpp.test.JSONCommunicatorTest): Text '2016.03.28T07:16:11.988Z' could not be parsed at index 4
````

**Comment by TVolden** [comment]
Got Travis CI working again (wierd gralde permission problem). You probably have to merge in my changes to .travis.yml to make it pass.

**Comment by kevinraddatz** [comment]
Seemd like somehow I missed this test :thinking:

**Pull request merged**